optuna/_hypervolume/utils.py

```diff
@@ -1,27 +1,26 @@
 import numpy as np
 
 
-def _compute_2d(solution_set: np.ndarray, reference_point: np.ndarray) -> float:
+def _compute_2d(sorted_pareto_sols: np.ndarray, reference_point: np.ndarray) -> float:
     """Compute the hypervolume for the two-dimensional space.
 
     This algorithm divides a hypervolume into
     smaller rectangles and sum these areas.
 
     Args:
-        solution_set:
+        sorted_pareto_sols:
             The solution set which we want to compute the hypervolume.
+            Note that this set is assumed to be sorted by the first value and Pareto optimal.
+            As this set is Pareto optimal and sorted by the first order, the second value
+            monotonically decreasing.
         reference_point:
             The reference point to compute the hypervolume.
     """
-    assert solution_set.shape[1] == 2 and reference_point.shape[0] == 2
-    if not np.isfinite(reference_point).all():
+    assert sorted_pareto_sols.shape[1] == 2 and reference_point.shape[0] == 2
+    if not np.all(np.isfinite(reference_point)):
         return float("inf")
```

Comment:
How about validating this branch?

Reply:
Removed here because it is covered by wfg.

Followup: Make _compute_2d a static method to prevent any external access.